runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/MemoryMonitor.java

```diff
@@ -222,15 +248,18 @@ static MemoryMonitor forTest(
       boolean canDumpHeap,
       double gcThrashingPercentagePerPeriod,
       @Nullable String uploadToGCSPath,
-      File localDumpFolder) {
+      File localDumpFolder,
+      @Nullable Duration jfrProfileDuration) {
     return new MemoryMonitor(
         gcStatsProvider,
         sleepTimeMillis,
         shutDownAfterNumGCThrashing,
         canDumpHeap,
         gcThrashingPercentagePerPeriod,
         uploadToGCSPath,
-        localDumpFolder);
+        localDumpFolder,
+        "test-worker",
```

Comment:
You might as well allow for this parameter to be passed in and then you can also validate that the filename contains the worker id.